NCMP_for_2-1-1_and_2-2-1/WHO_code_test.R

```diff
@@ -0,0 +1,104 @@
+###############################################################################################
+#
+# The user must set the working directory to match the directory of this R script
+# setwd() # uncomment this line to set the directory of the downloded script,, + csv files, keeping the folder with the data one directory above
+#
+#
+#NB: The WHO function needs to be run on each year of data separately - so 1 csv file per year
+###############################################################################################
+
+# install the package and load it in your R session:
+#install.packages("anthro",dependencies=TRUE, type="win.binary") #only need to do this once
+library(anthro) #need to run every time you start a new session, if you want to use the package
+
+
+# reading the data from 2018-19 download of the zipped file from NHS Digital, after extracting
+# assuming the data csv is one level up from the current working directory
+NCMP_data <- read.csv("ncmp_1819_final_non_disclosive_published.csv")
+
+
+# rounding/truncating the age in months variable
+NCMP_data$AgeInMonths <- trunc(NCMP_data$AgeInMonths) 
+
+# clean up suppressions from dataset (not necessary when working with full data set)
+clean_NCMP <- NCMP_data[NCMP_data$Height > 0, ]
+
+# retaining only age groups of interest (under 5s, exclusing 5)
+clean_NCMP <- clean_NCMP[clean_NCMP$AgeInMonths < 60, ]
+
+# removing rows with suppressed imd 
+clean_NCMP <- nrow(clean_NCMP[clean_NCMP$suppress_imd != "Y", ])
+unique(clean_NCMP$schoolindexofmultipledepriv) # checking the range is as expected (1:10)
+
+# check how many entries were lost after all the cleaning:
+nrow(clean_NCMP) - nrow(NCMP_data) #quite a lot! But this is just for testing purposes, raw data should need no cleaning (?)
+(nrow(NCMP_data)-nrow(clean_NCMP))/nrow(NCMP_data)*100 # as above, in %
+
+decile_to_quintile <- data.frame(deecile = c(1:10), quintile = c(rep(1:5, each = 2)))
+clean_NCMP <- left_join
+# regroup imd variable from deciles into quintiles (pair in twos)
+# inefficient (?) nesting method - overwrite 2s with 1, 4s and 3s with 2, 5s and 6s with 3, 7s and 8s with 4, and 9s and 10s with 5
+clean_NCMP$schoolindexofmultipledepriv <- ifelse(clean_NCMP$schoolindexofmultipledepriv==1, 1, 
```

Comment:
Just realised an even easier way to redo this in one line!! Divide the deciles by two and round with ceiling (got to test it, but should work)! 

`clean_NCMP$school_imd_quintile <- ceiling(clean_NCMP$schoolindexofmultipledepriv/2)`

Reply:
Nice! Now why didn't I think of that

Reply:
I think we can just keep the pull request open. Not sure what the proper thing to do is though.